nixos/modules/installer/cd-dvd/channel.nix

```diff
@@ -31,6 +31,7 @@ let
 in
 
 {
+  nix.registry.nixpkgs.flake = (import (pkgs.path + "/flake.nix")).outputs { self.outPath = pkgs.path; };
```

Comment:
A simpler version is following:
`nix.registry.nixpkgs.flake.outPath = toString pkgs.path;`

Reply:
That's pretty ingenious. I like it.